### ISSUE TITLE
[Feature request] add gamma and color correction from old color spaces like Dolphin

### ISSUE THREAD

**Issue by Filoppi** [rt64/rt64#36]
See [https://github.com/dolphin-emu/dolphin/pull/11850](https://github.com/dolphin-emu/dolphin/pull/11850).
My guess is that Majora's Mask was developed on CRT TVs that followed the NTSC-M standard, with a gamma of about ~2.35. For the game to look as it was meant to in 2000, but on 2024 displays, gamma should be adjusted to ~2.2 and the color space from NTSC-J to Rec.709/sRGB.
*I have no proof of this, I have no insight on the game's development process, but it should be a solid guess.

Moved from [https://github.com/Zelda64Recomp/Zelda64Recomp/issues/270](https://github.com/Zelda64Recomp/Zelda64Recomp/issues/270). See the original ticket for more.

**Comment by DarioSamo** [comment]
> For the game to look as it was meant to in 2000, but on 2024 displays, gamma should be adjusted to ~2.2 and the color space from NTSC-J to Rec.709/sRGB. I have no proof of this, I have no insight on the game's development process, but it should be a solid guess.

This seems very subjective in an era when CRTs had wildly varying calibration and not many people could tell exactly what an exact reference is. I think showing the unfiltered output in this case is a good default and what most people are already used to. The shader already supports [gamma adjust](https://github.com/rt64/rt64/blob/48e2adeebf6e613d07421797347bae78eb50f9e6/src/shaders/VideoInterfacePS.hlsl#L18) and it's currently used in games that specify they're in linear color space on the VI instead of sRGB. At best I could see exposing an option to tweak this gamma value, but I think the default behavior is what most people will expect out of the output.

**Comment by Filoppi** [comment]
It's relatively subjective, but it's also true that Rec.709 (sRGB) very likely wasn't the color space the game was designed on and for.
This is why dolphin lets the users pick between each region's color space.
I think that how the game is currently being seen by people on modern displays theoretically has "random" colors that are not the ones the game was meant for.

That gamma adjustment in the shader is a secondary gamma curve around a power of 1, it's a different thing. Or at least, it seems to be based on that code alone?

> it's currently used in games that specify they're in linear color space on the VI instead of sRGB
Games could output in linear space on the N64? I don't understand :D (sorry I don't know this stuff). How could they then display correctly, and most importantly, without banding give the low bit depth?
sRGB gamma encoding very likely wasn't ever used in N64 games, but again, I'm happy to be proven wrong.

**Comment by DarioSamo** [comment]
> Games could output in linear space on the N64? I don't understand :D (sorry I don't know this stuff). How could they then display correctly, and most importantly, without banding give the low bit depth?

There's a few games that did like Star Fox 64. It is pretty infamous for showing overly dark on emulators that ignored this.

![image](https://github.com/rt64/rt64/assets/538504/b1176643-511f-48de-ab94-6c6136579a47)

The gamma value in the shader basically does nothing for most games except for the ones that specify this in the VI.

**Comment by Filoppi** [comment]
Right. Just a guess, but I don't think the SF 64 outputted "linear space" colors, they just relied on the N64 gamma correction function to change the overall brightness, but I doubt the pow value was as high as 2.2 or 2.4, maybe it was like 1.3? 1.5?
So I don't think this is exactly related. After that gamma correction, the game would have still been intended to be displayed on gamma 2.35 TVs, and thus we should convert it to gamma 2.2 to display it correctly on modern displays.
And "lighting" was likely not in sRGB space, or at least, the final look of the game wasn't, as no TVs from the era followed sRGB. Monitors did, but N64 games targeted TVs.

**Comment by sologuitarplaying** [comment]
Not opposed to this feature but I don't believe most games had a specific intended gamma. From what I understand it varied from set to set even after calibration, shifted with the brightness setting of the screen, and was not linear.

**Comment by Filoppi** [comment]
@sologuitarplaying no maybe they didn't deliberately target a very specific gamma, but from what I know the average gamma on CRT TVs was 2.35 (like between 2.3 and 2.4), so correcting for data would still be going towards the target direction.
Just because there's no perfect solution or no definitive answer as to what color space or gamma each game targeted, it doesn't mean we shouldn't do anything about it and show them in a likely "wrong" color space and gamma. Going towards what was the most common standards for the time is still better IMO.
For example, on Dolphin we suggested to try to use the color space of the region the game was developed on. Japanese TVs back then had a white point (temperature) a lot bluer than EU or USA.

**Comment by sologuitarplaying** [comment]
@Filoppi I'm not sure modern displays necessarily betray the creative intent at all (for gamma at least). 2.35 may have been the average but it's the average of a range from 2.1 to 2.6. Similar deal with color space, we can guess but we don't know. So I'm mixed about it. 

  There's also a lot of other differences in how these games looked at the time compared to how they look on modern displays, so there's a question of where you draw the line there.

**Comment by Filoppi** [comment]
@sologuitarplaying the line I drew with our Dolphin features is "emulate the color space/gamma differences, but none of the defects of CRT TVs"
